pkg/uefi/genericregion.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2018 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package uefi
+
+// GenericRegion represents the Region in the firmware.
+type GenericRegion struct {
+	// holds the raw data
+	buf []byte
+	//Metadata for extraction and recovery
+	ExtractPath string
+	// This is a pointer to the FlashRegion struct laid out in the ifd
+	flashRegion *FlashRegion
+	// Region Type as per the IFD
+	RegionType FlashRegionType
+}
+
+// SetFlashRegion sets the flash region
+func (gr *GenericRegion) SetFlashRegion(fr *FlashRegion) {
+	gr.flashRegion = fr
+}
+
+// FlashRegion gets the flash region
+func (gr *GenericRegion) FlashRegion() (fr *FlashRegion) {
+	return gr.flashRegion
+}
+
+// NewGenericRegion creates a new region
+func NewGenericRegion(buf []byte, r *FlashRegion, rt FlashRegionType) (Region, error) {
+	gr := &GenericRegion{buf: buf, flashRegion: r, RegionType: rt}
+
+	return gr, nil
+}
+
+// Type returns the flash region type
```

Comment:
Put a period at the end of this sentence.

Reply:
Done